Src/Test.UnitTests.Sarif.PatternMatcher.Cli/AnalyzeCommandTests.cs

```diff
@@ -529,5 +742,50 @@ private static string GetSingleLineRuleDefinition()
 
             return JsonConvert.SerializeObject(definitions);
         }
+
+        private static string GetSingleLineRuleDefinitionFailureLevel(FailureLevel level)
```

Comment:
>GetSingleLineRuleDefinitionFailureLevel [](http://example.com/codeflow?start=30&length=39)

This method generates the shared strings and json files needed to support end to end testing of static and dynamic validation. These files will then be returned by the mocked file system during the relevant tests. This allowed for the cleanest repro of the conditions observed in the reported bug.